libraries/chain/finality/finalizer.cpp

```diff
@@ -154,8 +151,13 @@ void my_finalizers_t::maybe_update_fsi(const block_state_ptr& bsp, const valid_q
 
    bool updated = false;
    for (auto& f : finalizers) {
-      if (has_voted_strong(bsp->active_finalizer_policy->finalizers, received_qc, f.first)) {
+      if (has_voted_strong(bsp->active_finalizer_policy->finalizers, received_qc.active_policy_sig, f.first)) {
          updated |= f.second.maybe_update_fsi(bsp);
+      } else if (bsp->pending_finalizer_policy) {
+         assert(received_qc.pending_policy_sig);
```

Comment:
I don't think it should be an `assert` here, maybe:

```suggestion
      } else if (bsp->pending_finalizer_policy && received_qc.pending_policy_sig) {
```

Reply:
As in the above case, the received qc should be as expected and should have already been validated. But I'll add an `EOS_ASSERT` to be paranoid.